EIPS/eip-3135.md

```diff
@@ -0,0 +1,203 @@
+
+The provider can stop serving when payer no longer send new valid payment message or the credit is bigger than deposit balance. The user can stop signing new message if provider is not available or is cheating. The provider can claim token on chain at any time using the message contains the largest credit number.
+
+Let's explain the usage of exclusive claimable token by two cases.
+
+### 1. Token as recharge card of general store
+
+Like ordinary recharge card, the user pays with money/cryptocurrency to store in advance for a recharge card (with bonus or discount). Then user "activate recharge card" by calling deposit function. When checking out, the user just sign a message with updated credit (old credit + consumption this time) to store server. Server then verify this message and check the credit off chain. Then the shopping process goes on and on with out any blockchain involved. Until the user want to refund. The shopping service should first make claim call to remove all credit from deposit balance, then call withdraw function to return remaining deposit to user. Finally the user should transfer all token back to store to get money back, which is not a part of this discussion. The advantages are as follows:
+
+- Token as card point can transfer between users freely
+- The functions are built in token contract so the interacts with blockchain can be minimal for both parties
+- The modification work from ordinary card system to token as card system is much less
+
+### 2. Token as counter of online service
+
+Assume we have a distributed VPN service which user pays by token. Unlike general store case, users and servers do not know each other and don't trust each other. So there is two ways to implement claimable token based on the value of this token. If this token is widely accepted and holds value, token itself will serve as currency to buy service; if this token is just created on demand and is nothing more than number, this token will serve as a contract which user can deposit other token to. When using VPN service, the user deposit some token, and establish a counter channel with VPN service miner. For each x MB network traffic, the user signs a message to service miner, miner checks credit and user's deposit balance (this check can be done by pool rather than single miner) and decides to serve or not. This checking/serving process is in parallel. Because user can't trust VPN server, the withdraw function must be called by user. To make sure user can't just use service and withdraw deposit back before server claims, some checks are needed, for example, user can only withdraw token when epoch is not increased in a month. So, the server should claim at least monthly. The advantages are as follows:
+
+- Token is equal to service, and can be transferred
+- Token can act as stock of other valuable token
+- Token consumes only when server servers, no trust needed
+- Interaction with blockchain is minimal
+
```

Comment:
Example use cases are more appropriate for Motivation section rather than Rationale section.